contracts/facets/Market/MarketFacet.sol

```diff
@@ -197,53 +196,57 @@ contract MarketFacet is IMarket {
         }
     }
 
-    function _handleAuction(uint256 _tokenID, Iutils.Bid calldata _bid, address _owner, address _creator)
+    function _handleAuction(uint256 _tokenID, address _tokenAddress ,address _owner, address _creator, Iutils.Bid calldata _bid)
         internal
     returns (bool){
         LibMarketStorage.MarketStorage storage ms = LibMarketStorage.marketStorage();
         // Manage if the Bid is of Auction Type
-        address lastBidder = ms._tokenAsks[_tokenID]._bidder;
+
+        Iutils.Ask storage askInfo = ms._tokenAsks[_tokenAddress][_owner][_tokenID];
+        address lastBidder = askInfo._bidder;
 
         require(
-            ms._tokenAsks[_tokenID]._firstBidTime == 0 ||
+            askInfo._firstBidTime == 0 ||
                 block.timestamp <
-                ms._tokenAsks[_tokenID]._firstBidTime +
-                    ms._tokenAsks[_tokenID]._duration,
+                askInfo._firstBidTime +
+                askInfo._duration,
             "Market: Auction expired"
         );
 
         require(
             _bid._amount >=
-                ms._tokenAsks[_tokenID]._highestBid +
-                    (ms._tokenAsks[_tokenID]._highestBid *
-                        (ms.minBidIncrementPercentage * ms.EXPO)) /
-                    (ms.BASE),
-            "Market: Must send more than last bid by minBidIncrementPercentage amount"
+                askInfo._highestBid +
+                    (askInfo._highestBid *
+                        (ms._minBidIncrementPercentage * ms._EXPO)) /
+                    (ms._BASE),
+            "Market: Must send more than last bid by _minBidIncrementPercentage amount"
         );
-        if (ms._tokenAsks[_tokenID]._firstBidTime == 0) {
+        if (askInfo._firstBidTime == 0) {
             // If this is the first valid bid, we should set the starting time now.
-            ms._tokenAsks[_tokenID]._firstBidTime = block.timestamp;
+            askInfo._firstBidTime = block.timestamp;
             // Set New Bid for the Token
```

Comment:
Remove comment

Reply:
we are setting the firstBidTime and the comment says same.